### ISSUE TITLE
'-l' option should not return all probes if a more selective option is also specified

### ISSUE THREAD

**Issue by tyroguru** [iovisor/bpftrace#499]
Currently, if you want to list the USDT probes available in a given process you specify the '-l' and -'p' flag. 

```
#  bpftrace -l -p 1597900 |wc -l
39372
```

However, this process only has 14 USDT probes:

```
# tplist.py -p 1597900 | wc -l
14
```

That's not particularly helpful. You can grep for them but you have to know what they are apriori (and also we relabel the provider to 'usdt' so you won't find your provider name anyway if you grep for it!).

Maybe the best way would be to provide the full option if only a `-l` option was provided and, if another flag is provided that gives a more restrictive selection (i.e., `-p`) then just provide was explicitly asked for.

**Comment by amscanne** [comment]
Is this still something to be done? The searching and listing semantics seemed to have changed a lot since the issue was filed, and now you can filter (which I think was the original request). E.g. `bpftrace -l -p 1597900 usdt:*` I believe satisfies the original issue was asking? Or maybe I'm misunderstanding. (@jordalgo for context)

**Comment by danobi** [comment]
Yeah a lot of work has gone into listing probes since this issue was filed. I'd probably say what we have is good enough

**Comment by jordalgo** [comment]
It's probably fine to close but let's give @tyroguru a few days to respond and see if he wants to update the task.

**Comment by tyroguru** [comment]
Things are a lot better now w.r.t probe listing so let's close this.
